providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithClientConfigurer.java

```diff
@@ -0,0 +1,167 @@
+package dev.openfeature.contrib.providers.flagsmith;
+
+import com.flagsmith.FlagsmithClient;
+import com.flagsmith.config.FlagsmithCacheConfig;
+import com.flagsmith.config.Retry;
+import dev.openfeature.contrib.providers.flagsmith.exceptions.InvalidCacheOptions;
+import dev.openfeature.contrib.providers.flagsmith.exceptions.InvalidOptions;
+
+/**
+ * FlagsmithClientConfigurer helps set up and validate the options for the FlagsmithClient
+ * used by the FlagsmithProvider class.
+ */
+public class FlagsmithClientConfigurer {
+
+    /**
+     * Check the options that have been provided to see if there are any issues.
+     * Exceptions will be thrown if there are issues found with the options.
+     *
+     * @param options the options used to create the provider
+     */
+    public static void validateOptions(FlagsmithProviderOptions options) {
```

Comment:
do we need this validation here, why do we need them as separate method? should flagmisth sdk validate them when initialising client?
if you still want to use it make it private and call it from initializeProvider()?